iceoryx-sys/src/sample.rs

```diff
@@ -0,0 +1,218 @@
+    #[must_use]
+    #[inline]
+    pub fn slice_from_raw_parts(sample: RawSample<T>, len: usize) -> RawSample<[T]> {
+        // SAFETY: `self.as_payload_ptr` returns a non-null ptr
+        unsafe { Self::new_unchecked(slice::from_raw_parts(sample.as_payload_ptr(), len)) }
+    }
+
+    /// Returns the length of a non-null raw slice.
+    #[must_use]
+    #[inline]
+    pub fn len(self) -> usize {
+        // SAFETY: `self.as_payload_ptr` returns a non-null ptr
+        unsafe { (*self.as_payload_ptr()).len() }
+    }
+}
+
+impl<T: ?Sized> Clone for RawSample<T> {
+    #[inline(always)]
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+
+impl<T: ?Sized> Copy for RawSample<T> {}
+
+impl<T: ?Sized> fmt::Debug for RawSample<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::Pointer::fmt(&self.as_payload_ptr(), f)
+    }
+}
+
+impl<T: ?Sized> fmt::Pointer for RawSample<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::Pointer::fmt(&self.as_payload_ptr(), f)
+    }
+}
+
+/// A `*mut T` non-zero sample pointer to the user payload.
+#[repr(transparent)]
+pub struct RawSampleMut<T: ?Sized> {
+    payload: *mut T,
+}
+
+impl<T: ?Sized> RawSampleMut<T> {
+    /// Creates a new `RawSampleMut`.
+    ///
+    /// # Safety
+    ///
+    /// `payload` must be non-null.
+    #[inline]
+    pub unsafe fn new_unchecked(payload: *mut T) -> Self {
+        debug_assert!(
+            !payload.is_null(),
+            "RawSampleMut::new_unchecked requires that the payload pointer is non-null"
+        );
+        Self { payload }
+    }
+
+    /// Creates a new `RawSampleMut`.
+    #[allow(clippy::not_unsafe_ptr_arg_deref)] // false positive
+    #[inline]
+    pub fn new(payload: *mut T) -> Option<Self> {
+        if !payload.is_null() {
+            // SAFETY: `payload` pointer is checked to be non-null
+            Some(unsafe { Self::new_unchecked(payload) })
+        } else {
+            None
+        }
+    }
+
+    /// Casts to a `RawSampleMut` of another type.
+    #[must_use]
+    #[inline]
+    pub fn cast<U>(self) -> RawSampleMut<U> {
+        // SAFETY: `self.as_payload_mut_ptr` returns a non-null ptr
+        unsafe { RawSampleMut::new_unchecked(self.as_payload_mut_ptr().cast::<U>()) }
+    }
+
+    /// Acquires the underlying payload pointer as `*const` pointer.
+    #[must_use]
+    #[inline(always)]
+    pub fn as_payload_ptr(self) -> *const T {
+        self.as_payload_mut_ptr()
+    }
+
+    /// Acquires the underlying payload pointer as `*mut` pointer.
+    #[must_use]
+    #[inline(always)]
+    pub fn as_payload_mut_ptr(self) -> *mut T {
+        self.payload
+    }
+
+    /// Returns a reference to the `ChunkHeader`.
+    #[must_use]
+    #[inline]
+    pub fn chunk_header(&self) -> &ChunkHeader {
+        // SAFTEY: `self.as_payload_ptr` returns a non-null ptr
+        unsafe { ChunkHeader::from_user_payload_unchecked(self.as_payload_ptr() as *const c_void) }
```

Comment:
You could use `.cast()`, see https://doc.rust-lang.org/std/primitive.pointer.html#method.cast instead of `as *const c_void`.  It is a bit better to read and since I discovered it I use it everywhere.

Reply:
Hehe, I was thinking the same until I notice that `cast` does a `as _` under the hood and then got lazy :sweat_smile: